### ISSUE TITLE
mime/quotedprintable: accept invalid quoted-printable as unquoted

### ISSUE THREAD

**Issue by cention-sany** [golang/go#13219]
golang version: 1.5.1

There are some emails use the equal sign in-appropriately without escape it with =3D. This will make golang library fail to parse this kind of email. But some mail clients such as Thunderbird able can parse these malformed emails.

If [this line](https://github.com/golang/go/blob/master/src/mime/quotedprintable/reader.go#L110) change to `break` then more emails can be successfully parsed. Or return certain error which indicate it is safe to bypass it.

**Comment by bradfitz** [comment]
Please provide a sample of such an invalid input.

**Comment by cention-sany** [comment]
Attached. This attachment (was *.eml) can be opened with Thunderbird but when parse through Golang, will give the error: quotedprintable: invalid hex byte 0x22

[sample.txt](https://github.com/golang/go/files/33043/sample.txt)

**Comment by rsc** [comment]
Relevant bits below. Looks like "Microsoft CDO for Exchange 2000" basically just lies about quoted printable.

```
MIME-Version: 1.0
X-Mailer: Microsoft CDO for Exchange 2000
Importance: Normal
Priority: normal
Content-Class: urn:content-classes:message
Content-Type: multipart/alternative;
	boundary="----=_NextPart_000_6A3C_01D11B6E.BC80DF90"

This is a multi-part message in MIME format.

------=_NextPart_000_6A3C_01D11B6E.BC80DF90
Content-Transfer-Encoding: quoted-printable
Content-Type: text/plain;
	charset="iso-8859-1"

<P><p><font face="Arial, Helvetica, sans-serif" size="2" style="font-size:13.5px">
------=_NextPart_000_6A3C_01D11B6E.BC80DF90
Content-Transfer-Encoding: quoted-printable
Content-Type: text/html

<HTML>
<BODY>
<DIV><br>
Hej igen!</DIV>
</BODY></HTML
------=_NextPart_000_6A3C_01D11B6E.BC80DF90--
```

**Comment by cention-sany** [comment]
I encountered another bad email which need [these lines](https://github.com/golang/go/blob/master/src/mime/quotedprintable/reader.go#L115-L116) to be commented out too.

In fact, I noticed many of Golang stdlib (mostly on email parsing as I am using these frequently) return error once they encounter error and do not try whether it can do something else. I reported once which was closed https://github.com/golang/go/issues/13448. After that actually I found there are many of them too. Another example will be [this](https://github.com/golang/go/blob/master/src/net/textproto/reader.go#L480), why must return? why not finish all first? and return everything that you find good as well as the error - some special error to indicate there are still some useful data. User of this library can then decide whether to ignore the error or not and take whatever good.

I understand this is not some kind error but the fact is the Golang stdlib returning nil or insufficient data once it encounters error (some avoidable or recoverable errors) forced me to move away from stdlib because the need to create compatible email client as Thunderbird using Golang. Still, I believe Golang can do better.
